conda/core/python_dist.py

```diff
@@ -0,0 +1,1103 @@
+    elif ".egg-info" in anchor_file:
+        sp_reference = basename(dirname(anchor_file))
+        dist_file = join(prefix_path, win_path_ok(dirname(anchor_file)))
+        dist_cls = PythonEggInfoDistribution
+        package_type = PackageType.SHADOW_PYTHON_EGG_INFO_DIR
+    else:
+        raise NotImplementedError()
+
+    pydist = None
+
+    # An egg-link might reference a folder where egg-info is not available
+    if dist_file is not None:
+        try:
+            pydist = dist_cls(dist_file)
+        except Exception as error:
+            print('get_python_distribution_info', error)
+
+    return pydist, sp_reference, package_type
+
+
+def get_python_record(anchor_file, prefix_path, context):
+    """
+    Convert a python package defined by an anchor file (Metadata information)
+    into a conda prefix record object.
+
+    Return `None` if the python record cannot be created.
+    """
+    # TODO: ensure that this dist is actually the dist that matches conda-meta
+    pydist, sp_reference, package_type = get_python_distribution_info(anchor_file, prefix_path)
+
+    if pydist is None:
+        return None
+
+    pypi_name = pydist.norm_name
+    conda_name = pypi_name_to_conda_name(pypi_name)
+
+    # TODO: Handle packages with different names (graphviz vs python-graphviz)
+    channel_name = 'pypi' if conda_name == pypi_name else 'pypi:' + pypi_name
+    channel = Channel(channel_name)
+    build = 'pypi_0'
+
+    if package_type == PackageType.SHADOW_PYTHON_EGG_INFO_FILE:
+        paths_data = None
+    elif package_type in (PackageType.SHADOW_PYTHON_DIST_INFO,
+                          PackageType.SHADOW_PYTHON_EGG_INFO_DIR):
+        paths_data = pydist.get_paths_data()
+    elif package_type == PackageType.SHADOW_PYTHON_EGG_LINK:
+        paths_data = pydist.get_paths_data()
+        channel = Channel('<develop>')
+        build = 'dev_0'
+    else:
+        raise NotImplementedError()
+
+    # TODO: This is currently adding additional conda dependencies for graphviz
+    # only, but other packages need something similar. This info should (could)
+    # be on the 'external_requirements', but that field is free form.
+    dependencies = pydist.get_conda_dependencies(context)
+    extra_dependencies = PYPI_CONDA_DEPS.get(pypi_name, [])
+    all_dependencies = list(dependencies) + extra_dependencies
+    print('conda:{0} (pypi:{1})'.format(conda_name, pypi_name))
+    for dependency in sorted(all_dependencies):
+        print('\t{}'.format(dependency))
+    print('\n')
+
+    python_rec = PrefixRecord(
+        package_type=package_type,
+        name=conda_name,
+        version=pydist.version,
+        channel=channel,
+        subdir='pypi',
+        fn=sp_reference,
+        build=build,
+        build_number=0,
+        paths_data=paths_data,
+        depends=all_dependencies,
+    )
+
+    return python_rec
+
+
+def get_python_records(anchor_files, prefix_path, python_version):
```

Comment:
Move this function to the top, or as close to the top as it makes sense (i.e. following all the global variable definitions, etc.).